config/300-pubsubsource.yaml

```diff
@@ -43,15 +43,20 @@ spec:
         metadata:
           type: object
         spec:
+          # TODO: update the OpenAPI to be much more robust.
+          # TODO: document required fields.
           properties:
-            gcpCredsSecret:
+            secret:
               type: object
-              description: "Credential to use to poll the GCP PubSub Subscription. It is not used to create or delete the Subscription, only to poll it. The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys)."
-            googleCloudProject:
+              # TODO: document we default.
+              description: "Credential to use to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it. The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys)."
```

Comment:
Could've sworn I jotted it down, but I guess not. Should we spell out exactly what permissions the credential needs to have?

Reply:
https://github.com/GoogleCloudPlatform/cloud-run-events/issues/15